Common/src/element_linear.cpp

```diff
@@ -621,1526 +203,260 @@ su2double CQUAD4::ComputeArea(void){
-      for (iDim = 0; iDim < nDim; iDim++) {
-        GradNi_Xj_Ref = 0.0;
-        GradNi_Xj_Curr = 0.0;
-        for (jDim = 0; jDim < nDim; jDim++) {
-          GradNi_Xj_Ref += Jac_Ref[iDim][jDim]*dNiXj[iNode][jDim];
-          GradNi_Xj_Curr += Jac_Curr[iDim][jDim]*dNiXj[iNode][jDim];
-        }
-        GaussPoint[iGauss]->SetGradNi_Xj(GradNi_Xj_Ref, iDim, iNode);
-        GaussPoint[iGauss]->SetGradNi_xj(GradNi_Xj_Curr, iDim, iNode);
-      }
-    }
+    dNiXj[iGauss][0][1] = -0.125*(1.0-Xi)*(1.0-Zeta);
+    dNiXj[iGauss][1][1] = -0.125*(1.0+Xi)*(1.0-Zeta);
+    dNiXj[iGauss][2][1] =  0.125*(1.0+Xi)*(1.0-Zeta);
+    dNiXj[iGauss][3][1] =  0.125*(1.0-Xi)*(1.0-Zeta);
+    dNiXj[iGauss][4][1] = -0.125*(1.0-Xi)*(1.0+Zeta);
+    dNiXj[iGauss][5][1] = -0.125*(1.0+Xi)*(1.0+Zeta);
+    dNiXj[iGauss][6][1] =  0.125*(1.0+Xi)*(1.0+Zeta);
+    dNiXj[iGauss][7][1] =  0.125*(1.0-Xi)*(1.0+Zeta);
+
+    /*--- dN/d zeta ---*/
+
+    dNiXj[iGauss][0][2] = -0.125*(1.0-Xi)*(1.0-Eta);
+    dNiXj[iGauss][1][2] = -0.125*(1.0+Xi)*(1.0-Eta);
+    dNiXj[iGauss][2][2] = -0.125*(1.0+Xi)*(1.0+Eta);
+    dNiXj[iGauss][3][2] = -0.125*(1.0-Xi)*(1.0+Eta);
+    dNiXj[iGauss][4][2] =  0.125*(1.0-Xi)*(1.0-Eta);
+    dNiXj[iGauss][5][2] =  0.125*(1.0+Xi)*(1.0-Eta);
+    dNiXj[iGauss][6][2] =  0.125*(1.0+Xi)*(1.0+Eta);
+    dNiXj[iGauss][7][2] =  0.125*(1.0-Xi)*(1.0+Eta);
+
+  }
+
+  /*--- Store the extrapolation functions ---*/
+
+  su2double ExtrapCoord[8][3], sqrt3 = 1.732050807568877;
+
+  ExtrapCoord[0][0] = -sqrt3;  ExtrapCoord[0][1] = -sqrt3;  ExtrapCoord[0][2] = -sqrt3;
+  ExtrapCoord[1][0] =  sqrt3;  ExtrapCoord[1][1] = -sqrt3;  ExtrapCoord[1][2] = -sqrt3;
+  ExtrapCoord[2][0] =  sqrt3;  ExtrapCoord[2][1] =  sqrt3;  ExtrapCoord[2][2] = -sqrt3;
+  ExtrapCoord[3][0] = -sqrt3;  ExtrapCoord[3][1] =  sqrt3;  ExtrapCoord[3][2] = -sqrt3;
+  ExtrapCoord[4][0] = -sqrt3;  ExtrapCoord[4][1] = -sqrt3;  ExtrapCoord[4][2] =  sqrt3;
+  ExtrapCoord[5][0] =  sqrt3;  ExtrapCoord[5][1] = -sqrt3;  ExtrapCoord[5][2] =  sqrt3;
+  ExtrapCoord[6][0] =  sqrt3;  ExtrapCoord[6][1] =  sqrt3;  ExtrapCoord[6][2] =  sqrt3;
+  ExtrapCoord[7][0] = -sqrt3;  ExtrapCoord[7][1] =  sqrt3;  ExtrapCoord[7][2] =  sqrt3;
+
+  for (iNode = 0; iNode < nNodes; iNode++) {
+    Xi = ExtrapCoord[iNode][0];
+    Eta = ExtrapCoord[iNode][1];
+    Zeta = ExtrapCoord[iNode][2];
+
+    NodalExtrap[iNode][0] = 0.125*(1.0-Xi)*(1.0-Eta)*(1.0-Zeta);
+    NodalExtrap[iNode][1] = 0.125*(1.0+Xi)*(1.0-Eta)*(1.0-Zeta);
+    NodalExtrap[iNode][2] = 0.125*(1.0+Xi)*(1.0+Eta)*(1.0-Zeta);
+    NodalExtrap[iNode][3] = 0.125*(1.0-Xi)*(1.0+Eta)*(1.0-Zeta);
+    NodalExtrap[iNode][4] = 0.125*(1.0-Xi)*(1.0-Eta)*(1.0+Zeta);
+    NodalExtrap[iNode][5] = 0.125*(1.0+Xi)*(1.0-Eta)*(1.0+Zeta);
+    NodalExtrap[iNode][6] = 0.125*(1.0+Xi)*(1.0+Eta)*(1.0+Zeta);
+    NodalExtrap[iNode][7] = 0.125*(1.0-Xi)*(1.0+Eta)*(1.0+Zeta);
   }
 
 }
 
-su2double CHEXA1::ComputeVolume(void){
+su2double CHEXA8::ComputeVolume(const FrameType mode){
```

Comment:
@rsanfer would it be worth computing the volume the "finite element way" instead of using this "finite volume way"? I think these methods are only used by the mesh deformation but nevertheless it would be the generic way to do it (I have no idea about the performance - accuracy trade-off here though).